### PR TITLE
Zend validator unit tests

### PR DESCRIPTION
**Pull request by rms230**
This pull request is more unit tests to improve the code coverage of zend-validator.

### CODE REVIEW

**Comment by gianarb** [comment]
It's good IMO :+1:

**Comment by Maks3w** [comment]
We are introducing a new way in the architecture of test that implies:

* test API methods 
* Use more data providers

The proposal made here do some assertions about non API methods which make the test less resilient 

My idea is take the test examples proposed here and include them in v3

**Comment by gianarb** [comment]
This PR increase the value of this library because it cover a lot of possibile problem and it works now with this version..

I don't understand why this update can be a problem

File: test/BitwiseTest.php [hunk 1]
```diff
@@ -177,4 +177,53 @@ public function testSetStrict()
...
+        $strict = true;
+        $validator = new Bitwise(0x1, Bitwise::OP_AND, $strict);
+        $this->assertEquals($strict, $validator->getStrict());
+    }
+
+    public function testIsValidWithInvalidOperator()
+    {
+        $validator = new Bitwise(0x1, 'or', false);
+        $expectedResult = false;
+        $this->assertEquals($expectedResult, $validator->isValid(0x2));
```
Comment (weierophinney) [inline test/BitwiseTest.php]: I'd argue you should use either `assertSame($expectedResult, $validator->isValid(...))` or, more simply, `assertFalse($validator->isValid(...))`. I can make that change on merge, however.

File: test/CallbackTest.php [hunk 2]
```diff
@@ -116,4 +117,22 @@ public function optionsCallback($value)
         $this->assertContains('something', $args);
         return $args;
     }
+
+    public function testIsValidWithNoCallback()
```
Comment (weierophinney) [inline test/CallbackTest.php]: Rename this to indicate what *behavior* you're testing: 'testIsValidRaisesExceptionInAbsenceOfCallback'. I can make that change on merge.

File: test/Db/AbstractDbTest.php [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+ * @group      Zend_Validator
+ */
+class AbstractDbTest extends \PHPUnit_Framework_TestCase
+{
+
+    protected $validator;
+
+    public function setUp()
+    {
+        $this->validator = new ConcreteDbValidator([
```
Comment (weierophinney) [inline test/Db/AbstractDbTest.php]: A better approach would be to use mocking for the DB adapter. I'll see if I can work up an example.

**Comment by weierophinney** [comment]
This patch improves the library *now*, and will help pave the way for v3. I'm scheduling it for the next bugfix release.

**Pull request closed without merging**